src/lib/providers/BidProvider.tsx

```diff
@@ -78,12 +86,27 @@ const BidProvider: React.FC<BidProviderType> = ({ children }) => {
 	const context = useWeb3React();
 	const { addNotification } = useNotification();
 	const contracts = useZnsContracts();
+	const zAuctionContract = useZnsContracts()?.zAuction;
 
-	const acceptBid = async (bidId: string) => {
+	const acceptBid = async (bidData: Bid) => { 
+		
+		if (!zAuctionContract) {
+			throw Error(`no contract`);
+		}
 		try {
-			await new Promise((resolve) => setTimeout(resolve, 2000));
-			addNotification(`Bid accepted`);
-			return;
+
+			const tx = await zAuctionContract?.acceptBid(
```

Comment:
You could use the `tryFunction` helper here that you've used in other hooks to simplify this code